### PR TITLE
Added feature for Web Application Firewal and link it to AGW and its listener

### LINKED ISSUE #353: Application Gateway WAF policy
To support resource "azurerm_web_application_firewall_policy" in CAF modules.

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/web_application_firewall_policy

### LINKED ISSUE #394: App Service module is failing in release 5.3.1
the app service module is failing when using the following:
- hashicorp/azurerm 2.55.0
- aztfmod/caf/azurerm 5.3.1

Capture

### PR DESCRIPTION
**Pull request by iriahk89**
resolve #353 
fixed #394

### CODE REVIEW

**Comment by arnaudlh** [comment]
@iriahk89 can you please review and make sure its in the example and in caf solution lz? 

From the CI run, looks like var declaration is missing to execute the scenario: 
```
The root module does not declare a variable named
"application_gateway_waf_policies" but a value was found in file
"/__w/terraform-azurerm-caf/terraform-azurerm-caf/examples/app_gateway/102-waf-policy/waf.tfvars".
To use this value, add a "variable" block to the configuration.
```
thanks

File: modules/networking/application_gateway/application_gateway.tf [hunk 1]
```diff
@@ -34,7 +34,7 @@ resource "azurerm_application_gateway" "agw" {
   zones              = try(var.settings.zones, null)
   enable_http2       = try(var.settings.enable_http2, true)
   tags               = try(local.tags, null)
-  firewall_policy_id = try(var.settings.firewall_policy_id, null)
```
Comment (LaurentLesle) [inline modules/networking/application_gateway/application_gateway.tf:37]: can you keep the firewall_policy_id for backward compatibility?

File: examples/app_gateway/102-waf-policy/waf.tfvars [hunk 2]
```diff
@@ -0,0 +1,22 @@
+application_gateway_waf_policies = {
```
Comment (LaurentLesle) [inline examples/app_gateway/102-waf-policy/waf.tfvars:1]: Try to have an example that covers all the blocks in the implementation. At least the public example https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/web_application_firewall_policy#example-usage

**Review by LaurentLesle — CHANGES_REQUESTED** [review]
Added view comments. Thanks @iriahk89

File: modules/networking/application_gateway/application_gateway.tf (hunk 1 above)
Comment (iriahk89) [reply modules/networking/application_gateway/application_gateway.tf:37]: Fixed

File: examples/app_gateway/102-waf-policy/waf.tfvars (hunk 2 above)
Comment (iriahk89) [reply examples/app_gateway/102-waf-policy/waf.tfvars:1]: Updated examples as advised

**Review by LaurentLesle — APPROVED** [review]

**Pull request merged**